src/net.cpp

```diff
@@ -1323,41 +1323,38 @@ void CConnman::SocketEvents(std::set<SOCKET> &recv_set, std::set<SOCKET> &send_s
     FD_ZERO(&fdsetSend);
     FD_ZERO(&fdsetError);
     SOCKET hSocketMax = 0;
-    bool have_fds = false;
 
     std::set<SOCKET> recv_select_set, send_select_set, error_select_set;
     GenerateSelectSet(recv_select_set, send_select_set, error_select_set);
 
+    if (recv_select_set.empty() && send_select_set.empty() && error_select_set.empty()) {
+        interruptNet.sleep_for(std::chrono::milliseconds(50));
+        return;
+    }
+
     for (SOCKET hSocket : recv_select_set) {
         FD_SET(hSocket, &fdsetRecv);
-        have_fds = true;
         hSocketMax = std::max(hSocketMax, hSocket);
     }
 
     for (SOCKET hSocket : send_select_set) {
         FD_SET(hSocket, &fdsetSend);
-        have_fds = true;
         hSocketMax = std::max(hSocketMax, hSocket);
     }
 
     for (SOCKET hSocket : error_select_set) {
         FD_SET(hSocket, &fdsetError);
-        have_fds = true;
         hSocketMax = std::max(hSocketMax, hSocket);
     }
 
-    int nSelect = select(have_fds ? hSocketMax + 1 : 0,
-                         &fdsetRecv, &fdsetSend, &fdsetError, &timeout);
+    int nSelect = select(hSocketMax + 1, &fdsetRecv, &fdsetSend, &fdsetError, &timeout);
 
     if (nSelect == SOCKET_ERROR)
     {
-        if (have_fds)
-        {
-            int nErr = WSAGetLastError();
-            LogPrintf("socket select error %s\n", NetworkErrorString(nErr));
-            for (unsigned int i = 0; i <= hSocketMax; i++)
-                FD_SET(i, &fdsetRecv);
-        }
+        int nErr = WSAGetLastError();
+        LogPrintf("socket select error %s\n", NetworkErrorString(nErr));
+        for (unsigned int i = 0; i <= hSocketMax; i++)
+            FD_SET(i, &fdsetRecv);
```

Comment:
I see this is the same as the code currently, but it seems like this should be `fdsetError` instead of `fdsetRecv`, even though they are handled exactly the same.

Reply:
I don't think it should be there at all really. select() failures dont mean the sockets themselves have failed, but i guess it could mean we made a mistake and called select() with a closed socket?

Reply:
Yeah, then iterating through each one and calling `recv` would identify which peer to disconnect. Though, if it's `fdsetError` and not `fdsetRecv`, it probably shouldn't disconnect when `nBytes == 0`.